### PR TITLE
Dịch "bptt_vn" - Phần 6

### LINKED ISSUE #1595: Dịch "bptt_vn" - Phần 6
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_recurrent-neural-networks/bptt_vn.md#L343-L393

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by cuongvng**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1595

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 1]
```diff
@@ -354,7 +354,14 @@ Later on in :numref:`chap_modern_rnn` we will see how more sophisticated sequenc
 In practice, this truncation is effected by *detaching* the gradient after a given number of steps.
 -->
 
-*dịch đoạn phía trên*
+Nhiều điều có thể rút ra từ biểu thức khá đáng sợ này.
```
Comment (cuongvng) [inline chapter_recurrent-neural-networks/bptt_vn.md:357]:
```suggestion
Nhiều điều có thể rút ra từ biểu thức phức tạp này.
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 2]
```diff
@@ -354,7 +354,14 @@ Later on in :numref:`chap_modern_rnn` we will see how more sophisticated sequenc
 In practice, this truncation is effected by *detaching* the gradient after a given number of steps.
 -->
 
-*dịch đoạn phía trên*
+Nhiều điều có thể rút ra từ biểu thức phức tạp này.
+Đầu tiên, các kết quả trung gian được lưu lại, đó là các luỹ thừa của $\mathbf{W}_{hh}$ khi tính các số hạng của hàm mất mát $L$.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/bptt_vn.md:358]:
```suggestion
Đầu tiên, lưu các kết quả trung gian lại sẽ rất có lợi, đó là các luỹ thừa của $\mathbf{W}_{hh}$ khi tính các số hạng của hàm mất mát $L$.
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 3]
```diff
@@ -354,7 +354,14 @@ Later on in :numref:`chap_modern_rnn` we will see how more sophisticated sequenc
...
+Thứ hai, ví dụ tuyến tính đơn giản này cho thấy một vài vấn đề chính của các mô hình chuỗi dài: chúng bao gồm các luỹ thừa rất lớn của $\mathbf{W}_{hh}^j$.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/bptt_vn.md:359]:
```suggestion
Thứ hai, ví dụ tuyến tính đơn giản này mà đã cho thấy một vài vấn đề chủ chốt của các mô hình chuỗi dài: chúng tiềm ẩn chứa các luỹ thừa rất lớn của $\mathbf{W}_{hh}^j$.
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 4]
```diff
@@ -388,7 +398,12 @@ Formalize this statement.
 4. Besides gradient clipping, can you think of any other methods to cope with gradient explosion in recurrent neural networks?
 -->
 
-*dịch đoạn phía trên*
+1. Giả sử ta có ma trận đối xứng $\mathbf{M} \in \mathbb{R}^{n \times n}$ có các trị riêng $\lambda_i$.
+Không mất tính tổng quát, giả sử chúng được sắp xếp theo thứ tự tăng dần $\lambda_i \leq \lambda_{i+1}$.
+Chứng minh rằng $\mathbf{M}^k$ có các trị riêng là $\lambda_i^k$.
+2. Chứng minh rằng với vector bất kì $\mathbf{x} \in \mathbb{R}^n$, xác suất cao là $\mathbf{M}^k \mathbf{x}$ sẽ tương ứng với trị riêng lớn nhất $\mathbf{v}_n$ của $\mathbf{M}$. <!-- không hiểu đoạn này lắm, hóng cao nhân :) -->
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/bptt_vn.md:404]:
```suggestion
2. Chứng minh rằng với vector bất kì $\mathbf{x} \in \mathbb{R}^n$, xác suất cao là $\mathbf{M}^k \mathbf{x}$ sẽ gần như là cùng phương với vector trị riêng lớn nhất $\mathbf{v}_n$ của $\mathbf{M}$.
```
M là đối xứng nên có thể phân tách eigen thành M^k=U @ Lambda^k @ U^Transpose.
Lambda là diagonal matrix có eigenvector ở đường chéo. Vì các trị riêng này được mũ k lần lên giá trị nào lớn nhất sẽ dominate các giá trị khác (exploding), các giá trị < 1 mũ k lần lên là gần =0 luôn (vanishing)

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 5]
```diff
@@ -388,7 +398,12 @@ Formalize this statement.
...
+3. Kết quả trên áp dụng vào gradient của mạng nơ-ron truy hồi sẽ như thế nào?
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/bptt_vn.md:405]:
```suggestion
3. Kết quả trên có ý nghĩa như thế nào khi tính gradient của mạng nơ-ron truy hồi?
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 6]
```diff
@@ -354,7 +354,14 @@ Later on in :numref:`chap_modern_rnn` we will see how more sophisticated sequenc
 In practice, this truncation is effected by *detaching* the gradient after a given number of steps.
 -->
 
-*dịch đoạn phía trên*
+Nhiều điều có thể rút ra từ biểu thức phức tạp này.
+Đầu tiên, lưu các kết quả trung gian lại sẽ rất có lợi, đó là các luỹ thừa của $\mathbf{W}_{hh}$ khi tính các số hạng của hàm mất mát $L$.
+Thứ hai, ví dụ tuyến tính đơn giản này mà đã cho thấy một vài vấn đề chủ chốt của các mô hình chuỗi dài: chúng tiềm ẩn chứa các luỹ thừa rất lớn của $\mathbf{W}_{hh}^j$.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/bptt_vn.md:359]:
```suggestion
Thứ hai, ví dụ tuyến tính đơn giản này mà đã cho thấy một vài vấn đề chủ chốt của các mô hình chuỗi dài: chúng tiềm ẩn các luỹ thừa rất lớn của $\mathbf{W}_{hh}^j$.
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 7]
```diff
@@ -354,7 +354,14 @@ Later on in :numref:`chap_modern_rnn` we will see how more sophisticated sequenc
...
+Trong đó, khi $j$ lớn, các trị riêng nhỏ hơn $1$ sẽ tiêu biến, còn các trị riêng lớn hơn $1$ sẽ phân kì.
+Các mô hình này có các giá trị số không ổn định, dẫn đến việc quan trọng hoá quá mức các chi tiết không liên quan trong quá khứ.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/bptt_vn.md:361]:
```suggestion
Các mô hình này không có tính ổn định số học, dẫn đến việc chúng quan trọng hoá quá mức các chi tiết không liên quan trong quá khứ. 
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 8]
```diff
@@ -354,7 +354,14 @@ Later on in :numref:`chap_modern_rnn` we will see how more sophisticated sequenc
 In practice, this truncation is effected by *detaching* the gradient after a given number of steps.
 -->
 
-*dịch đoạn phía trên*
+Nhiều điều có thể rút ra từ biểu thức phức tạp này.
+Đầu tiên, lưu các kết quả trung gian lại sẽ rất có lợi, đó là các luỹ thừa của $\mathbf{W}_{hh}$ khi tính các số hạng của hàm mất mát $L$.
+Thứ hai, ví dụ tuyến tính đơn giản này mà đã cho thấy một vài vấn đề chủ chốt của các mô hình chuỗi dài: chúng tiềm ẩn chứa các luỹ thừa rất lớn của $\mathbf{W}_{hh}^j$.
+Trong đó, khi $j$ lớn, các trị riêng nhỏ hơn $1$ sẽ tiêu biến, còn các trị riêng lớn hơn $1$ sẽ phân kì.
+Các mô hình này có các giá trị số không ổn định, dẫn đến việc quan trọng hoá quá mức các chi tiết không liên quan trong quá khứ. 
+Một cách giải quyết vấn đề này là bỏ bớt các số hạng trong tổng ở mức độ thuận tiện cho tính toán.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/bptt_vn.md:362]:
```suggestion
Một cách giải quyết vấn đề này là bỏ bớt các số hạng trong tổng ở một mức độ thuận tiện cho việc tính toán.
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 9]
```diff
@@ -354,7 +354,14 @@ Later on in :numref:`chap_modern_rnn` we will see how more sophisticated sequenc
...
+Sau đây tại :numref:`chap_modern_rnn`, ta sẽ thấy cách các mô hình chuỗi phức tạp như LSTM giải quyết vấn đề này tốt hơn.
+Trong lập trình, ta bỏ bớt các số hạng bằng cách *tách rời* gradient sau số bước xác định.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/bptt_vn.md:364]:
```suggestion
Trong lập trình, ta bỏ bớt các số hạng bằng cách *tách rời* gradient sau một số bước xác định.
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 10]
```diff
@@ -369,7 +376,10 @@ In practice, this truncation is effected by *detaching* the gradient after a giv
 * For efficient computation, intermediate values are cached.
 -->
 
-*dịch đoạn phía trên*
+* Lan truyền ngược theo thời gian là cách áp dụng lan truyền ngược cho các mô hình chuỗi có trạng thái ẩn.
+* Việc cắt bỏ là cần thiết để thuận tiện tính toán và ổn định các giá trị số.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/bptt_vn.md:380]:
```suggestion
* Việc cắt bỏ là cần thiết để thuận tiện cho việc tính toán và ổn định các giá trị số.
```

File: chapter_recurrent-neural-networks/bptt_vn.md [hunk 11]
```diff
@@ -369,7 +376,10 @@ In practice, this truncation is effected by *detaching* the gradient after a giv
...
+* Luỹ thừa lớn của ma trận có thể làm các trị riêng tiêu biến hoặc phân kì. Điều này được thể hiện trong hiện tượng tiêu biến hoặc bùng nổ gradient.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/bptt_vn.md:381]:
```suggestion
* Luỹ thừa lớn của ma trận có thể làm các trị riêng tiêu biến hoặc phân kì. Điều này được thể hiện dưới hiện tượng tiêu biến hoặc bùng nổ gradient.
```
